only apply fix_path for href attrib in <link>s

johndoh opened the pull request:
for #9941

I could not think of a use case for the `href` of an `<a>` to go via `static.php` so I split up the existing regex to allow special handling of the `href` attribute and take account of the tag its being used it.

It could be that this is a bit too complicated and really we only need to fix the path when we are in a `<link>` tag. W3Schools says `href` is only valid in `<a>`, `<area>`, `<base>` and `<link>` tags. So the new pattern could be changed to specifically match `<link ` rather than ignore `<a `.

johndoh committed:
skip fix_path for href attrib in <a>s

johndoh committed:
add very basic test for fix_paths()

alecpl commented:
Another option might be to change the `resource_location()` method that it ignores URLs that (after removing everything after `?`) do not point to a file (must end with `.something`). What do you think?

johndoh committed:
add tests for external urls

johndoh commented:
Not in `resource_location()` because `file_callbac()` would still add the skin path to the URLs. The root cause is because `?` was added to the pattern for `fix_paths()` so now both `file_callback()`  and then `resource_location()` get called in more cases than before.

Doing something with the existence of a file extension in `file_callback()` would solve the basic case but I think a 3rd party skin or a plugin could add <a>'s with links to anywhere it wants and putting those hrefs through post processing like this is unnecessary. I have added a couple of tests for external links which I think this code should not alter.

Rather than the "not an <a>" pattern I have added here I think adding a "only hrefs in <link>s" would be enough.

alecpl commented:
Let's do "only hrefs in `<link>`".

johndoh committed:
apply fix_paths() to <link>s only

johndoh committed:
PHP 8.5: Remove redundant setAccessible() calls
